pkg/controllers/awsloadbalancercontroller/service.go

```diff
@@ -84,6 +91,13 @@ func (r *AWSLoadBalancerControllerReconciler) updateService(ctx context.Context,
 		updatedService.Spec.Type = desired.Spec.Type
 	}
 
+	for annotationKey, annotationValue := range desired.Annotations {
+		if currentAnnotationValue, ok := updatedService.Annotations[annotationKey]; !ok || currentAnnotationValue != annotationValue {
+			updatedService.Annotations = desired.Annotations
+			break
+		}
+	}
```

Comment:
So, it's possible to have unsolicited annotations but only until any desired one is changed.
Not very strict policy against unsolicited annotations: they are accepted at the edit time but may be gone anytime after.

Maybe a strict policy like `DeepEqual` check of current vs desired instead (like for the containers in the deployment)? Or reset only the desired ones?

Reply:
That's true, I should not reset the annotations instead add it to the existing list. Faulty logic. Annotations can be added by third party controllers and we don't want to erase them.